# cspell:ignore semseg
/**:
  ros__parameters:

    # refine segmentation mask by overlay roi class
    # disable when sematic segmentation accuracy is good enough
    is_roi_overlap_segment: true

    # minimum existence_probability of detected roi considered to replace segmentation
    overlap_roi_score_threshold: 0.3

    # publish color mask for result visualization
    is_publish_color_mask: false

    roi_overlay_segment_label:
      UNKNOWN : true
      CAR : false
      TRUCK : false
      BUS : false
      MOTORCYCLE : true
      BICYCLE : true
      PEDESTRIAN : true
      ANIMAL: true

    image_path: "$(var data_path)/tensorrt_yolov10/000000000307.jpg" 
    model_path: "$(var data_path)/tensorrt_yolov10/$(var model_name).onnx" # The onnx file name for YOLOX model.
    label_path: "$(var data_path)/tensorrt_yolov10/label.txt" # The label file path for YOLOX model.
    color_map_path: "$(var data_path)/tensorrt_yolov10/semseg_color_map.csv"
    score_threshold: 0.35 # Objects with a score lower than this value will be ignored. This threshold will be ignored if specified model contains EfficientNMS_TRT module in it.
    nms_threshold: 0.7 # Detection results will be ignored if IoU over this value. This threshold will be ignored if specified model contains EfficientNMS_TRT module in it.

    precision: "fp16" # Operation precision to be used on inference. Valid value is one of: [fp32, fp16, int8].
    calibration_algorithm: "MinMax" # Calibration algorithm to be used for quantization when precision==int8. Valid value is one of: [Entropy, (Legacy | Percentile), MinMax].
    dla_core_id: -1 # If positive ID value is specified, the node assign inference task to the DLA core.
    quantize_first_layer: false # If true, set the operating precision for the first (input) layer to be fp16. This option is valid only when precision==int8.
    quantize_last_layer: false # If true, set the operating precision for the last (output) layer to be fp16. This option is valid only when precision==int8.
    profile_per_layer: false # If true, profiler function will be enabled. Since the profile function may affect execution speed, it is recommended to set this flag true only for development purpose.
    clip_value: 0.0 # If positive value is specified, the value of each layer output will be clipped between [0.0, clip_value]. This option is valid only when precision==int8 and used to manually specify the dynamic range instead of using any calibration.
    preprocess_on_gpu: true # If true, pre-processing is performed on GPU.
    gpu_id: 0 # GPU ID to select CUDA Device
    calibration_image_list_path: "" # Path to a file which contains path to images. Those images will be used for int8 quantization.
